src/client.js

```diff
@@ -305,427 +305,558 @@ class ConnectionParameters {
+        "select": {
+          "node": [
+            { "expr": "@id" },
+            { "expr": "@name" },
+            { "expr": "@label" },
+            { "expr": "@type" },
+            { "expr": "@account" },
+            { "expr": "@password" },
+            { "expr": "@server" }
+          ]
+        },
+        "where": {
+          "condition": [
+            { "expr": client.sdk.escapeXtk`@name=${extAccountName}` },
+            { "expr": "@type=3" }
+          ]
         }
-        throw CampaignException.CREDENTIALS_FOR_INVALID_EXT_ACCOUNT(extAccountName, type);
+      };
+        // Convert to current representation
+      queryDef = client._convertToRepresentation(queryDef, "SimpleJson");
+      const query = client.NLWS.xtkQueryDef.create(queryDef);
+      const extAccount = await query.executeQuery();
+
+      const cipher = await client._getSecretKeyCipher();
+      const type = EntityAccessor.getAttributeAsLong(extAccount, "type");
+      if (type == 3) {
+        // Mid-souring
+        const endpoint = EntityAccessor.getAttributeAsString(extAccount, "server");
+        const user = EntityAccessor.getAttributeAsString(extAccount, "account");
+        const password = cipher.decryptPassword(EntityAccessor.getAttributeAsString(extAccount, "password"));
+        return ConnectionParameters.ofUserAndPassword(endpoint, user, password, client._connectionParameters._options);
+      }
+      throw CampaignException.CREDENTIALS_FOR_INVALID_EXT_ACCOUNT(extAccountName, type);
     }
 
-}
+  }
 
-// ========================================================================================
-// ACC Client
-// ========================================================================================
+  // ========================================================================================
+  // File Uploader
+  // ========================================================================================
+
+  const fileUploader = (client) => {
+    /**
+         * Will call the SOAP method(IncreaseValue) to increament the counter
+         * @returns {Promise<number>}
+         */
+    const _increaseValue = async () => {
+      const xtkCounter= await client.NLWS.xtkCounter.increaseValue({name: 'xtkResource'});
+      return xtkCounter
+    }
+
+    /**
+         * This will create and return fileRes object
+         * @param counter
+         * @param data
+         * @returns {{originalName, internalName: string, fileName, useMd5AsFilename: string, xtkschema: string, storageType: number, label, md5: (string|string|string|null|any|number)}}
+         * @private
+         */
+    const _createFileRes = (counter, data) => {
+      return {
+        internalName: 'RES' + counter,
+        md5: data[0].md5,
+        label: data[0].fileName,
+        fileName: data[0].fileName,
+        originalName: data[0].fileName,
+        useMd5AsFilename: '1',
+        storageType: 5,
+        xtkschema: 'xtk:fileRes'
+
+      }
+    }
+
+    /**
```

Comment:
This comment is not really helpful. Is there a particular reason to have created separate functions for each of the API calls? I mean, what's the point of calling this._write(fileRes) vs. simply client.NLWS.xtkSession.write(fileRes) ? It seems adding a function here makes things most verbose and does not really help understand the intent of the code

Reply:
It was done to consider the separation of concerns.

Reply:
What do you mean?